chef-utils/lib/chef-utils/dsl/platform_family.rb

```diff
@@ -0,0 +1,282 @@
+#
+# Copyright:: Copyright 2018-2019, Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require_relative "../internal"
+
+module ChefUtils
+  module DSL
+    module PlatformFamily
+      include Internal
+
+      # Determine if the current node is arch linux.
+      #
+      # @param [Chef::Node] node
+      #
+      # @return [Boolean]
+      #
+      def arch?(node = __getnode)
+        node["platform_family"] == "arch"
+      end
+      # chef-sugar backcompat methods
+      alias_method :arch_linux?, :arch?
+
+      # Determine if the current node is aix
+      #
+      # @param [Chef::Node] node
+      #
+      # @return [Boolean]
+      #
+      def aix?(node = __getnode)
+        node["platform_family"] == "aix"
+      end
+
+      # Determine if the current node is a member of the debian family.
+      #
+      # @param [Chef::Node] node
+      #
+      # @return [Boolean]
+      #
+      def debian?(node = __getnode)
+        node["platform_family"] == "debian"
+      end
+
+      # Determine if the current node is a member of the fedora family.
+      #
+      # @param [Chef::Node] node
+      #
+      # @return [Boolean]
+      #
+      def fedora?(node = __getnode)
+        node["platform_family"] == "fedora"
+      end
+
+      # Determine if the current node is a member of the OSX family.
+      #
+      # @param [Chef::Node] node
+      #
+      # @return [Boolean]
+      #
+      def mac_os_x?(node = __getnode)
+        node["platform_family"] == "mac_os_x"
+      end
+      alias_method :osx?, :mac_os_x?
+      alias_method :mac?, :mac_os_x?
+      alias_method :macos?, :mac_os_x?
```

Comment:
Same with this one. macos? should be the real method name with the aliases for the old os x names